integration_test/reporting_dropped_events/reporting_dropped_events_test.go

```diff
@@ -543,6 +547,140 @@ func TestReportingDroppedEvents(t *testing.T) {
 	})
 }
 
+func TestReportingErrorIndexing(t *testing.T) {
```

Comment:
Please use a separate file for this test. We also need to verify all scenarios (stages) where error index should be populated. I can pick up this task

Reply:
Thanks for pointing this out, I was a bit hesitant to create a separate integration test, since it was similar to the test written for dropped events.
But it's better to have a separate integration test scenario for it.

Reply:
Added the necessary integration test here. 
https://github.com/rudderlabs/rudder-server/blob/40e3c25089e0832df9eeae9c2391d5355c1a9154/integration_test/reporting_error_index/reporting_error_index_test.go